File: Server/src/main/content/global/skill/production/crafting/data/PotteryData.java
package content.global.skill.production.crafting.data;

import core.game.node.item.Item;

public enum PotteryData {
    POT(new Item(1787), new Item(1931), 1, 6.3, 6.3),
    DISH(new Item(1789), new Item(2313), 7, 15, 10),
    BOWL(new Item(1791), new Item(1923), 8, 18, 15),
    PLANT(new Item(5352), new Item(5350), 19, 20, 17.5),
    LID(new Item(4438), new Item(4440), 25, 20, 20);

    private final Item unfinished;
    private final Item product;
    private final int level;
    private final double exp;
    private final double fireExp;

    PotteryData(Item unfinished, Item product, int level, double exp, double fireExp) {
        this.unfinished = unfinished;
        this.product = product;
        this.level = level;
        this.exp = exp;
        this.fireExp = fireExp;
    }

    public static PotteryData forId(int id) {
        for (PotteryData def : PotteryData.values()) {
            if (def.getUnfinished().getId() == id) {
                return def;
            }
        }
        return null;
    }

    public Item getUnfinished() {
        return unfinished;
    }
    public Item getProduct() {
        return product;
    }
    public int getLevel() {
        return level;
    }
    public double getExp() {
        return exp;
    }
    public double getFireExp() {
        return fireExp;
    }
}

File: Server/src/main/content/global/skill/production/crafting/data/PotteryData.kt
package content.global.skill.production.crafting.data

import core.api.consts.Items
import core.game.node.item.Item

/**
 * Pottery data
 *
 * @property unfinished Represents the unfinished state of the pottery item.
 * @property product Represents the final product of the pottery item.
 * @property level Indicates the level of the pottery item.
 * @property exp Represents the experience points associated with the pottery item.
 * @property fireExp Represents the experience points gained from firing the pottery item.
 * @constructor Pottery data Initializes the properties of the PotteryData enum.
 */
enum class PotteryData(
    val unfinished: Item, // Holds the unfinished item representation
    val product: Item,    // Holds the final product representation
    val level: Int,       // Indicates the level of the pottery item
    val exp: Double,      // Experience points for the pottery item
    val fireExp: Double    // Experience points gained from firing the pottery item
) {
    /**
     * Pot.
     */
    POT(
        unfinished = Item(Items.UNFIRED_POT_1787),
        product = Item(Items.EMPTY_POT_1931),
        level = 1,
        exp = 6.3,
        fireExp = 6.3
    ),

    /**
     * Dish.
     */
    DISH(
        unfinished = Item(Items.UNFIRED_PIE_DISH_1789),
        product = Item(Items.PIE_DISH_2313),
        level = 7,
        exp = 15.0,
        fireExp = 10.0
    ),

    /**
     * Bowl.
     */
    BOWL(unfinished = Item(Items.UNFIRED_BOWL_1791),
        product = Item(Items.BOWL_1923),
        level = 8, 18.0,
        fireExp = 15.0
    ),

    /**
     * Plant.
     */
    PLANT(
        unfinished = Item(Items.UNFIRED_PLANT_POT_5352),
        product = Item(Items.PLANT_POT_5350),
        level = 19,
        exp = 20.0,
        fireExp = 17.5
    ),

    /**
     * Lid.
     */
    LID(
        unfinished = Item(Items.UNFIRED_POT_LID_4438),
        product = Item(Items.POT_LID_4440),
        level = 25,
        exp = 20.0,
        fireExp = 20.0
    );

    companion object {
        @JvmStatic
        fun forId(id: Int): PotteryData? {
            for (def in values()) {
                if (def.unfinished.id == id) {
                    return def
                }
            }
            return null
        }
    }
}
